src/main/scala/beam/agentsim/agents/household/HouseholdFleetManager.scala

```diff
@@ -98,16 +101,63 @@ class HouseholdFleetManager(
     case GetVehicleTypes(triggerId) =>
       sender() ! VehicleTypesResponse(vehicles.values.map(_.beamVehicleType).toSet, triggerId)
 
-    case MobilityStatusInquiry(_, _, _, triggerId) =>
-      availableVehicles = availableVehicles match {
-        case firstVehicle :: rest =>
-          logger.debug("Vehicle {} is now taken", firstVehicle.id)
-          firstVehicle.becomeDriver(sender)
-          sender() ! MobilityStatusResponse(Vector(ActualVehicle(firstVehicle)), triggerId)
-          rest
-        case Nil =>
-          sender() ! MobilityStatusResponse(Vector(), triggerId)
-          Nil
+    case MobilityStatusInquiry(personId, whenWhere, _, requireVehicleCategoryAvailable, triggerId) =>
+      {
+        for {
+          neededVehicleCategory              <- requireVehicleCategoryAvailable
+          emergencyHouseholdVehicleGenerator <- maybeEmergencyHouseholdVehicleGenerator
+          vehicle                            <- emergencyHouseholdVehicleGenerator.createVehicle(personId, nextVehicleIndex, neededVehicleCategory)
```

Comment:
Should we create the vehicle here? This should probably be pushed into the if, because if there are available vehicles then we hit the creation code for nothing.

Reply:
Done